src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -242,30 +235,34 @@ where
             start_signal,
         );
 
-        source
-            .inner
-            .map(move |((output, result), from_time, diff)| {
-                let result = match result {
-                    Ok(msg) => Ok(SourceOutput {
-                        key: msg.key.clone(),
-                        value: msg.value.clone(),
-                        metadata: msg.metadata.clone(),
-                        from_time: from_time.clone(),
-                    }),
-                    Err(err) => Err(err.clone()),
-                };
-                ((*output, result), from_time.clone(), *diff)
-            })
-            .capture_into(PusherCapture(reclock_pusher));
+        for (id, export) in exports {
+            let (reclock_pusher, reclocked) = reclock(&remap_collection, config.as_of.clone());
```

Comment:
Here is the first substantial change of this PR. Instead of rendering a single reclock operator we iterate over all the exports and instantiate one reclock operator per output.